services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/UpdateHakemusITest.kt

```diff
@@ -0,0 +1,1039 @@
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.toUpdateRequest
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withArea
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withAreas
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withContractor
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withCustomer
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withCustomerWithContactsRequest
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withDates
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withInvoicingCustomer
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withName
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withRequiredCompetence
+import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withWorkDescription
+import fi.hel.haitaton.hanke.factory.HakemusyhteystietoFactory
+import fi.hel.haitaton.hanke.factory.HankeFactory
+import fi.hel.haitaton.hanke.factory.HankeKayttajaFactory
+import fi.hel.haitaton.hanke.findByType
+import fi.hel.haitaton.hanke.firstReceivedMessage
+import fi.hel.haitaton.hanke.hakemus.ApplicationContactType.HAKIJA
+import fi.hel.haitaton.hanke.hakemus.ApplicationContactType.TYON_SUORITTAJA
+import fi.hel.haitaton.hanke.logging.AuditLogRepository
+import fi.hel.haitaton.hanke.logging.ObjectType
+import fi.hel.haitaton.hanke.test.USERNAME
+import fi.hel.haitaton.hanke.toJsonString
+import fi.hel.haitaton.hanke.tormaystarkastelu.Autoliikenneluokittelu
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluTulos
+import io.mockk.checkUnnecessaryStub
+import io.mockk.clearAllMocks
+import io.mockk.confirmVerified
+import java.time.ZonedDateTime
+import java.util.UUID
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.NullSource
+import org.junit.jupiter.params.provider.ValueSource
+import org.springframework.beans.factory.annotation.Autowired
+
+class UpdateHakemusITest(
+    @Autowired private val hakemusService: HakemusService,
+    @Autowired private val hakemusRepository: HakemusRepository,
+    @Autowired private val hakemusyhteystietoRepository: HakemusyhteystietoRepository,
+    @Autowired private val hankeRepository: HankeRepository,
+    @Autowired private val auditLogRepository: AuditLogRepository,
+    @Autowired private val hakemusFactory: HakemusFactory,
+    @Autowired private val hankeFactory: HankeFactory,
+    @Autowired private val hankeKayttajaFactory: HankeKayttajaFactory,
+    @Autowired private val alluClient: AlluClient,
+) : IntegrationTest() {
+
+    companion object {
+        @JvmField
+        @RegisterExtension
+        val greenMail: GreenMailExtension =
```

Comment:
Is this needed in this test class?

Reply:
It's necessary here. It's used to test the notification emails when a käyttäjä is added to a hakemus.

It's not used in `HakemusServiceITest.kt`, so removing it from there.